### PR TITLE
[Inference API] Add Docs for Mistral Embedding Support for the Inference API

### PR DESCRIPTION
**Pull request by markjhoy**
Add docs for the Mistral embeddings task type (from https://github.com/elastic/elasticsearch/pull/109194)

Also updates the [Semantic Search tutorial](https://www.elastic.co/guide/en/elasticsearch/reference/master/semantic-search-inference.html) to add in Mistral embeddings as well.

### CODE REVIEW

File: docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc [hunk 1]
```diff
@@ -77,12 +83,12 @@ include::infer-api-ingest-pipeline.asciidoc[tag=azure-openai]
   </div>
   <div tabindex="0"
        role="tabpanel"
-       id="infer-api-ingest-azure-ai-studio-tab"
-       aria-labelledby="infer-api-ingest-azure-ai-studio"
+       id="infer-api-ingest-mistral-tab"
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc:86]: The widget for `azure-ai-studio` has been edited out. This is the reason why the tutorial page is not rendered correctly.

File: docs/reference/tab-widgets/inference-api/infer-api-reindex-widget.asciidoc [hunk 2]
```diff
@@ -27,8 +27,14 @@
...
-            id="infer-api-reindex-azure-openai">
+            aria-controls="infer-api-reindex-azure-ai-studio-tab"
+            id="infer-api-reindex-azure-ai-studio">
+      Azure AI Studio
+    </button>
+    <button role="tab"
+            aria-selected="false"
+            aria-controls="infer-api-reindex-mistral"
+            id="infer-api-reindex-mistral">
       Azure AI Studio
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-reindex-widget.asciidoc:38]: This line should be `Mistral`.

**Review by szabosteve — COMMENTED** [review]
I left a couple of comments to fix the widgets in the tutorial.

The IDE autoformatted the parameter descriptions. We follow this pattern:
```
`name_of_the_param`::
(Required or optional, <data type>)
Description.
```
In the autoformatted text, the description and the info between brackets are in the same line. Would it be possible to undo those changes?

File: docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc:86]: No problem at all, thank you for writing this content up!

**Comment by markjhoy** [comment]
> In the autoformatted text, the description and the info between brackets are in the same line. Would it be possible to undo those changes?

Sorry - just re-read this -- latest commit should fix many of these if not all

File: docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc [hunk 3]
```diff
@@ -66,8 +72,8 @@ include::infer-api-ingest-pipeline.asciidoc[tag=openai]
   </div>
   <div tabindex="0"
        role="tabpanel"
-       id="infer-api-ingest-azure-openai-tab"
-       aria-labelledby="infer-api-ingest-azure-openai"
+       id="infer-api-ingest-azure-ai-studio-tab"
+       aria-labelledby="infer-api-ingest-azure-ai-studio"
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc:76]: It is still not quite right. Now we miss the `Azure Open AI` tab. The following buttons and tabs need to be in this file:
* Cohere
* HuggingFace
* OpenAI
* Azure OpenAI (we miss this one now, my suggestion adds it back)
* Azure AI Studio
* Mistral
The buttons are great! The issue is in the second half of the file where we configure the tabs' content. We need a section that references the `azure-openai-tab` and includes the corresponding asciidoc file. Right now, this section from line 72 to line 82 contains references to `azure-ai-studio` but the include file (line 80) refers to `azure openai`. The suggestion solves that problem. However, we still need a section for `azure-ai-studio`. I know, the names are pretty confusing. :) 

```suggestion
       id="infer-api-ingest-azure-openai-tab"
       aria-labelledby="infer-api-ingest-azure-openai"
```

**Review by szabosteve — COMMENTED** [review]
Thank you for your edits, @markjhoy! I left one more comment about the ingest pipeline widget. Please let me know if I explained the issue poorly. Thanks again!

File: docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc (hunk 3 above)
Comment (markjhoy) [reply docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc:76]: Ohhhh - sorry! I saw that and thought to myself "that doesn't look right", so I intuitively changed it. Changing it back now. Thanks for the explanation.

Comment (markjhoy) [reply docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc]: Oh wait - hold up, there is a section in the infer-api-ingest-pipeline file for[ azure-ai-studio](https://github.com/elastic/elasticsearch/blob/fdb5058b13abbbc5ce3a5d7c0ab1a7878c72176c/docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline.asciidoc?plain=1#L116) -- is this what you think is missing here? If so, I think it's just a matter of making sure the `include::` references this (which, it does not at the moment...) let me try that and commit it and see...

Comment (markjhoy) [reply docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc:76]: @szabosteve - I _think_ my latest commit did indeed fix it properly (without the workaround) - let me know your thoughts!

Comment (markjhoy) [reply docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc:76]: (although it's not working properly for the reindexing steps examples... debugging this now)

Comment (szabosteve) [reply docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc:76]: @markjhoy The `azure-openai` tab content is still missing. After `openai` we need an `azure-openai` but now we just have an `azure-ai-studio`. To fix this, we need to add the following code after line 71:

```
</div>
  <div tabindex="0"
       role="tabpanel"
       id="infer-api-ingest-azure-openai-tab"
       aria-labelledby="infer-api-ingest-azure-openai"
       hidden="">
++++
include::infer-api-ingest-pipeline.asciidoc[tag=azure-openai]
++++
```

Comment (markjhoy) [reply docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc]: > The azure-openai tab content is still missing

Yep - I found that and I _think_ my latest commit should fix it (also, the "hidden" was missing from the HuggingFace portion for the "Create an inference endpoint" part - which caused it to be visible on first load as well... - fixed this in the commit too)... after the build is complete, hopefully all will be working as it should 🤞

Comment (markjhoy) [reply docs/reference/tab-widgets/inference-api/infer-api-ingest-pipeline-widget.asciidoc]: ... and so close... Mistral is missing from the index pipeline step, but other than that, everything else is working... another round incoming shortly...

File: docs/reference/tab-widgets/inference-api/infer-api-reindex-widget.asciidoc [hunk 4]
```diff
@@ -27,10 +27,16 @@
...
             aria-selected="false"
-            aria-controls="infer-api-reindex-azure-openai-tab"
-            id="infer-api-reindex-azure-openai">
+            aria-controls="infer-api-reindex-azure-ai-studio-tab"
+            id="infer-api-reindex-azure-ai-studio">
       Azure AI Studio
     </button>
+    <button role="tab"
+            aria-selected="false"
+            aria-controls="infer-api-reindex-mistral"
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-reindex-widget.asciidoc:36]: I think that will fix the missing Mistral content for the reindex step.
```suggestion
            aria-controls="infer-api-reindex-mistral-tab"
```

Comment (markjhoy) [reply docs/reference/tab-widgets/inference-api/infer-api-reindex-widget.asciidoc]: OK - I _think_ this is the last of it... 😮‍💨  thank you for all your reviews on this!

**Review by szabosteve — APPROVED** [review]
Yay, thanks for all the hard work on this one! LGTM!

**Comment by markjhoy** [comment]
> Yay, thanks for all the hard work on this one! LGTM!

Thank you! I found one last lingering issue - when I merged main in to resolve some conflicts, it removed the service settings for Mistral in the `put-inference` docs... this new commit should complete this.

**Comment by markjhoy** [comment]
> > Yay, thanks for all the hard work on this one! LGTM!
> 
> Thank you! I found one last lingering issue - when I merged main in to resolve some conflicts, it removed the service settings for Mistral in the `put-inference` docs... this new commit should complete this.

That all looks good! Going to merge. Thanks again.

**Pull request merged**